src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/EjpContactPreferenceFieldMidEventHandler.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo.NO;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo.YES;
+import static uk.gov.hmcts.reform.iacaseapi.domain.handlers.HandlerUtils.sourceOfAppealEjp;
+
+import java.util.Collections;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class EjpContactPreferenceFieldMidEventHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private static final String ADDRESS_PAGE_ID = "appellantAddress";
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return (callbackStage == PreSubmitCallbackStage.MID_EVENT
+                && callback.getEvent() == Event.START_APPEAL
+                && callback.getPageId().equals(ADDRESS_PAGE_ID)
+                && sourceOfAppealEjp(callback.getCaseDetails().getCaseData()));
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+
+        if (asylumCase.read(IS_LEGALLY_REPRESENTED_EJP, YesOrNo.class).orElse(NO).equals(NO)) {
+            asylumCase.clear(CONTACT_PREFERENCE);
+        } else if (asylumCase.read(IS_LEGALLY_REPRESENTED_EJP, YesOrNo.class).orElse(NO).equals(YES)) {
+            asylumCase.write(CONTACT_PREFERENCE_UNREP, Collections.emptyList());

```

Comment:
Do we need this now? I thought writing over at ABOUT_TO_SUBMIT should not need this mid-event.

Reply:
This mid event is still need to clear the contactPreference radio buttons and checkboxes. When I went to test it again those were still showing pre-populated